prow/cmd/horologium/main.go

```diff
@@ -93,9 +93,27 @@ func main() {
 		logrus.WithError(err).Fatal("Error starting config agent.")
 	}
 
-	prowJobClient, err := o.kubernetes.ProwJobClient(configAgent.Config().ProwJobNamespace, o.dryRun)
+	cfg, err := o.kubernetes.InfrastructureClusterConfig(o.dryRun)
 	if err != nil {
-		logrus.WithError(err).Fatal("Error getting Kubernetes client.")
+		logrus.WithError(err).Fatal("Failed to get prowjob kubeconfig")
+	}
+	cluster, err := cluster.New(cfg, func(o *cluster.Options) { o.Namespace = configAgent.Config().ProwJobNamespace })
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to construct prowjob client")
+	}
+	if _, err := cluster.GetCache().GetInformer(interrupts.Context(), &prowapi.ProwJob{}); err != nil {
```

Comment:
Is this just for sanity check? Or `GetInformer` actually does something like `SetInformer` in the `Cache`? Would be nice to document it imo

Reply:
Added some godocs